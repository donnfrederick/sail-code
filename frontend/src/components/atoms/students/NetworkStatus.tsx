import * as React from 'react'
import styled from 'styled-components'
import isHuawei from 'utils/isHuawei'
import resolvePath from 'utils/resolvePath'

type Status = 'good' | 'normal' | 'bad'

interface Props {
  status: Status
}

export default (props: Props) => {
  const { status } = props
  return status === 'good' ? (
    <Good>
      <Icon
        src={resolvePath.image('students/conversation/network-status@3x.png')}
      />GOOD
    </Good>
  ) : status === 'normal' ? (
    <Normal>
      <Icon
        src={resolvePath.image('students/conversation/network-status@3x.png')}
      />NORMAL
    </Normal>
  ) : status === 'bad' ? (
    <Bad>
      <Icon
        src={resolvePath.image('students/conversation/network-status@3x.png')}
      />BAD
    </Bad>
  ) : null
}

const Good = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 200px;
  height: 56px;
  border-radius: 28px;
  background-color: #00c7c0;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3);'
    : 'box-shadow: 0 6px 10px -4px rgba(5, 68, 102, 0.3),'} ${isHuawei()
    ? ''
    : '  0 0 12px 0 rgba(0, 0, 0, 0.2);'}
  font-size: 22px;
  font-weight: 500;
  line-height: 56px;
  letter-spacing: 0.5px;
  text-align: center;
  color: #ffffff;
`

const Normal = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 200px;
  height: 56px;
  border-radius: 28px;
  background-color: #e6b122;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3);'
    : 'box-shadow: 0 6px 10px -4px rgba(5, 68, 102, 0.3),'} ${isHuawei()
    ? ''
    : '  0 0 12px 0 rgba(0, 0, 0, 0.2);'}
  font-size: 22px;
  font-weight: 500;
  line-height: 56px;
  letter-spacing: 0.5px;
  text-align: center;
  color: #ffffff;
`

const Bad = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 200px;
  height: 56px;
  border-radius: 28px;
  background-color: #f75e66;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3);'
    : 'box-shadow: 0 6px 10px -4px rgba(5, 68, 102, 0.3),'} ${isHuawei()
    ? ''
    : '  0 0 12px 0 rgba(0, 0, 0, 0.2);'}
  font-size: 22px;
  font-weight: 500;
  line-height: 56px;
  letter-spacing: 0.5px;
  text-align: center;
  color: #ffffff;
`

const Icon = styled.img`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 18px;
  width: 24px;
  height: 24px;
  margin: auto;
`
