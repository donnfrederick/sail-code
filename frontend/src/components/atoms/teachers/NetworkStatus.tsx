import * as React from 'react'
import styled from 'styled-components'
import isHuawei from 'utils/isHuawei'

type Status = 'good' | 'normal' | 'bad'

interface Props {
  status: Status
}

export default (props: Props) => {
  const { status } = props
  return status === 'good' ? (
    <Good>回線状況 : 良</Good>
  ) : status === 'normal' ? (
    <Normal>回線状況 : 普通</Normal>
  ) : status === 'bad' ? (
    <Bad>回線状況 : 悪</Bad>
  ) : null
}

const Good = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 216px;
  height: 48px;
  border-radius: 24px;
  background-color: #00c7c0;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.2)'};
  font-size: 24px;
  font-weight: 500;
  line-height: 48px;
  letter-spacing: 0px;
  text-align: center;
  color: #ffffff;
`

const Normal = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 216px;
  height: 48px;
  border-radius: 24px;
  background-color: #e6b122;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.2)'};
  font-size: 24px;
  font-weight: 500;
  line-height: 48px;
  letter-spacing: 0px;
  text-align: center;
  color: #ffffff;
`

const Bad = styled.div`
  position: absolute;
  top: 32px;
  left: 32px;
  width: 216px;
  height: 48px;
  border-radius: 24px;
  background-color: #f75e66;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.2)'};
  font-size: 24px;
  font-weight: 500;
  line-height: 48px;
  letter-spacing: 0px;
  text-align: center;
  color: #ffffff;
`
