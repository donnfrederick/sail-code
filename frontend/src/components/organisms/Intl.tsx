import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
