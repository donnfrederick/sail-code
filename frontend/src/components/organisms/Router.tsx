import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
