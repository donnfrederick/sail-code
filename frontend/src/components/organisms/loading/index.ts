import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
