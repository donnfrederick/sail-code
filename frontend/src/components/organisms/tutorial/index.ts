import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
