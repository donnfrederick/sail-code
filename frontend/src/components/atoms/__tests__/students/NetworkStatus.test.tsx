import NetworkStatus from 'components/atoms/students/NetworkStatus'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <NetworkStatus />', () => {
  const tree = renderer
    .create(
      <Intl>
        <div>
          <NetworkStatus status="good" />
          <NetworkStatus status="normal" />
          <NetworkStatus status="bad" />
        </div>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
