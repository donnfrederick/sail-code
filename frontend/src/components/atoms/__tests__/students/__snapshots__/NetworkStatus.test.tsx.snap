// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <NetworkStatus /> 1`] = `
<div>
  <div
    className="sc-bdVaJa laRVbW"
  >
    <img
      className="sc-bxivhb fRSJqW"
      src="/assets/img/students/conversation/network-status@3x.png"
    />
    GOOD
  </div>
  <div
    className="sc-bwzfXH dIckEq"
  >
    <img
      className="sc-bxivhb fRSJqW"
      src="/assets/img/students/conversation/network-status@3x.png"
    />
    NORMAL
  </div>
  <div
    className="sc-htpNat lmnHMa"
  >
    <img
      className="sc-bxivhb fRSJqW"
      src="/assets/img/students/conversation/network-status@3x.png"
    />
    BAD
  </div>
</div>
`;
